### PR TITLE
[LC-891] Update jsonrpcclient version

### PR DESCRIPTION
**Pull request by metalg0su**
(coupled: https://github.com/icon-project/loopchain/pull/476)

# Goal
- Update version of `jsonrpcclient` library from 2.6.0 to 3.3.5 (latest)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,11 +11,11 @@ branches:
   - travis-test
 
 install:
-  - pip install -r requirements.txt
-  - make generate-proto
+  - python -m pip install --upgrade pip
```
Comment (zsaladin) [inline .travis.yml]: Is it essential?

Comment (metalg0su) [reply .travis.yml]: Sorry for temporary commit.

I have encountered a problem related to travis-ci.
I have no idea why unit tests are failed on travis whereas succeed on my local.

I'm trying to figure out problem by using the temporary commit.

Comment (metalg0su) [reply .travis.yml]: Added. 
Please check this commit and resolve if it is acceptable.

https://github.com/icon-project/loopchain/pull/476/commits/540346d7ad68e846819c2dd14d57930158f7ce5d

File: iconrpcserver/utils/json_rpc.py [hunk 2]
```diff
@@ -52,14 +52,15 @@ def process_response(self, response, log_extra=None, log_format=None):
             # Log the response before processing it
             self.log_response(response, log_extra, log_format)
             # If it's a json string, parse to object
-            if isinstance(response, basestring):
+            if isinstance(response, str) or isinstance(response, bytes):
```
Comment (zsaladin) [inline iconrpcserver/utils/json_rpc.py]: How about using `AnyStr`?
https://docs.python.org/3/library/typing.html#typing.AnyStr

File: iconrpcserver/utils/json_rpc.py [hunk 3]
```diff
@@ -52,14 +52,15 @@ def process_response(self, response, log_extra=None, log_format=None):
...
                 try:
                     response = json.loads(response)
                 except ValueError:
-                    raise exceptions.ParseResponseError()
+                    # FIXME: Tried to ReceivedErrorResponseError, but needs jsonrpcclient.response.ErrorResponse
+                    raise exceptions.JsonRpcClientError()
```
Comment (zsaladin) [inline iconrpcserver/utils/json_rpc.py:60]: Is there any reason why `ParseResponseError` is replaced with `JsonRpcClientError`?

File: iconrpcserver/utils/json_rpc.py (hunk 2 above)
Comment (yakkle) [inline iconrpcserver/utils/json_rpc.py]:
```suggestion
            if isinstance(response, (str, bytes)):
```

File: iconrpcserver/utils/json_rpc.py (hunk 3 above)
Comment (metalg0su) [reply iconrpcserver/utils/json_rpc.py:60]: `ParseResponseError` is deprecated in https://github.com/bcb/jsonrpcclient/commit/9638aa18efc84db8fed8c7ca8250150843149cf3

So i choose parent class of the exception, `JsonRpcClientError`. But I'm not sure this exception is acceptable in this line.

File: iconrpcserver/utils/json_rpc.py (hunk 2 above)
Comment (metalg0su) [reply iconrpcserver/utils/json_rpc.py]: Can you explain usage of `typing.AnyStr` in `isinstace` built-in method?

I tried:
```bash
Python 3.7.3 (default, Dec 10 2019, 01:42:03)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.10.2 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from typing import AnyStr

In [2]: a = "a"

In [3]: isinstance(a, AnyStr)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-a31d16311937> in <module>
----> 1 isinstance(a, AnyStr)

TypeError: isinstance() arg 2 must be a type or tuple of types

...

In [6]: b = b"b"

In [7]: isinstance(b, AnyStr)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-7-23dc579345ee> in <module>
----> 1 isinstance(b, AnyStr)

TypeError: isinstance() arg 2 must be a type or tuple of types
```

Comment (zsaladin) [reply iconrpcserver/utils/json_rpc.py]: Oops Sorry
In this case (bytes, str) is correct

Comment (metalg0su) [reply iconrpcserver/utils/json_rpc.py]: Thank you for your suggestion!
I applied yours.

File: tests/dispatcher/test_reception.py [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import pytest
-from asyncmock import AsyncMock
+from mock.mock import AsyncMock
```
Comment (yakkle) [inline tests/dispatcher/test_reception.py]:
```suggestion
from mock import AsyncMock
```

File: tests/dispatcher/test_ws_dispatcher.py [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import json
 
 import pytest
-from asyncmock import AsyncMock
-from jsonrpcclient.request import Request
+from mock.mock import AsyncMock
```
Comment (yakkle) [inline tests/dispatcher/test_ws_dispatcher.py]: ditto

File: requirements.txt [hunk 6]
```diff
@@ -1,18 +1,17 @@
...
-jsonrpcclient==2.6.0
 secp256k1==0.13.2
 sanic-cors==0.9.6
 earlgrey>=0.0.4
 iconcommons>=1.0.5
 pytest
 pytest-asyncio
-asyncmock
-pytest-mock
+mock
```
Comment (yakkle) [inline requirements.txt]: please specify mock version and move to above of `pytest`.

Comment (metalg0su) [reply requirements.txt]: Thank you.
This is the major problem that i suffered on travis-ci test failure.

I set mock version above 4.0.1 which is latest.

File: tests/dispatcher/test_reception.py (hunk 4 above)
Comment (metalg0su) [reply tests/dispatcher/test_reception.py]: Thank you~

File: iconrpcserver/utils/json_rpc.py (hunk 3 above)
Comment (zsaladin) [reply iconrpcserver/utils/json_rpc.py:60]: How about using `ReceivedErrorResponseError`?

Comment (metalg0su) [reply iconrpcserver/utils/json_rpc.py:60]: First time, I used `ReceivedErrorResponseError`, but this exception requires `ErrorResponse` to retrieve its message.

But I have no idea to reach `ErrorResponse` at this line. (Possible `str` or `bytes` response)

**Review by yakkle — APPROVED** [review]

**Review by zsaladin — APPROVED** [review]

File: iconrpcserver/utils/json_rpc.py (hunk 3 above)
Comment (metalg0su) [reply iconrpcserver/utils/json_rpc.py:60]: Thank you for your precious review.

We can discuss the replacement of `CustomAiohttpClient` later, in order to follow bare API of jsonrpcclient.

**Review by windies21 — APPROVED** [review]

**Pull request merged**